### DISCUSSION TITLE
feat: Data purging Parameters

### DISCUSSION THREAD

**Discussion by Zarkhator03** [open-webui/open-webui#3729] [Ideas]
## Description

This feature request aims to introduce granular control over chat lifetime and data purging parameters, both at the system-wide and user-specific levels. By providing configurable quotas, open chat limits, and chat lifetimes, we can minimize storage of unused data, optimize resource utilization, and improve overall system performance for use at scale.

## Proposed Features

- **UI Fields and Database:**
  - Add configurable fields to the user interface and database for:
    - **Quota:** Maximum allowed chat length (e.g., number of messages)
    - **Open Chats:** Maximum number of concurrently open chats per user
    - **Chat Lifetime:** Duration after which a chat is automatically deleted

- **Default Values:**
  - Establish default values for automatic deletion:
    - **Max Chat Length:** Define a standard maximum chat length
    - **Max Open Chats:** Set a default limit for concurrent open chats
    - **Chat Lifetime:** Specify a default duration for chat expiration
  - default values editable by administrators

- **User-Specific Overrides:**
  - Allow administrators to override default values for individual users, providing flexibility in managing specific user requirements or addressing resource constraints.
  - If multiple user roles are added in the future, one value can be selected per user role.

**Comment by tkg61** [comment]
Would love to have this as well. Also knowing how much data is currently being stored (how soon till you are full) would be great as well.

**Comment by chaser1802** [comment]
Would also love to see this, but also with ability to define retention schedules for other entities, e.g., prompts, feedback, knowledge sources, so that we can better support compliance requirements.

**Comment by Classic298** [comment]
Related: https://github.com/open-webui/open-webui/pull/16520
